scripts/deploy-frontend

```diff
@@ -11,12 +11,18 @@ fi
 
 set -euo pipefail # exit on error; treat unset variables as errors; exit on errors in piped commands
 SELF_DIR="$(perl -e 'use File::Basename; use Cwd "abs_path"; print dirname(abs_path(@ARGV[0]));' -- "$0")" # figure out the absolute path to the current script, regardless of pwd (perl is more cross-platform than realpath; https://stackoverflow.com/a/30795461)
+ROOT_DIR="$SELF_DIR/.."
```

Comment:
Is there a reason why we don't rely on plain `pwd`? I typically orchestrate everything to be run on the root project directory, but is there a reason why this wouldn't be the case? Do some hosting platforms do things differently?

Reply:
It's mostly for safety. Having a build script run with wrong assumptions can make a mess. So I prefer to either a) enforce the PWD or b) make it irrelevant.